javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -95,6 +73,19 @@ predicate isConstantBooleanReturnValue(Expr e) {
   isConstantBooleanReturnValue(e.(LogNotExpr).getOperand())
 }
 
+/**
+ * Holds if `e` is a defensive expression with a fixed outcome.
+ */
+predicate isConstantDefensive(Expr e) {
+  exists(Expr defensive |
+    defensive = e or
+    // traverse negations
+    defensive.(LogNotExpr).getOperand+() = e
```

Comment:
Doesn't this skip through arbitrary unary expression operands? Is this what we want?